projects/packages/external-connections/src/class-external-connections.php

```diff
@@ -0,0 +1,315 @@
+		$connect_url = $body->services->$service->connect_URL ?? null;
+		set_transient( $transient_name, $connect_url, DAY_IN_SECONDS );
+		return $connect_url;
+	}
+
+	/**
+	 * Retrieves a connection of the provided service.
+	 *
+	 * @param string $service The service identifier.
+	 * @return array|null The connection details, or `null` if no matching connection is found.
+	 */
+	public static function get_connection( $service ) {
+		if ( ( new Host() )->is_wpcom_simple() ) {
+			require_lib( 'external-media-service/external-media-list' );
+			require_lib( 'external-connections' );
+
+			$connections = \WPCOM_External_Connections::init();
+			$token       = \ExternalMediaService::get_service_token( $service, get_current_user_id() );
+
+			if ( ! empty( $token->unique_id ) ) {
+				return $connections->get_keyring_connection_item( $token->unique_id );
+			}
+		} else {
+			$response = Client::wpcom_json_api_request_as_user( '/me/connections' );
+			if ( is_wp_error( $response ) ) {
+				return null;
+			}
+			$body = json_decode( wp_remote_retrieve_body( $response ) );
+			if ( isset( $body->connections ) && is_array( $body->connections ) ) {
+				foreach ( $body->connections as $connection ) {
+					if ( $service === $connection->service ) {
+						return (array) $connection;
+					}
+				}
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Checks if a connection to the specified service exists and is active.
+	 *
+	 * @param string $service The service identifier.
+	 *
+	 * @return bool True if an active connection exists, false otherwise.
+	 */
+	public static function has_connection( $service ) {
+		$connection = self::get_connection( $service );
+
+		if ( empty( $connection ) ) {
+			return false;
+		}
+
+		$connection_status = $connection['status'] ?? 'not_connected';
+		return $connection_status === 'ok';
+	}
+
+	/**
+	 * Deletes a connection for the provided service.
+	 *
+	 * @param string $service The service identifier.
+	 * @return boolean Whether the connection was deleted.
+	 */
+	public static function delete_connection( $service ) {
+		$connection = self::get_connection( $service );
+		if ( empty( $connection ) ) {
+			return false;
+		}
+
+		if ( ( new Host() )->is_wpcom_simple() ) {
+			if ( get_current_user_id() === $connection['user_ID'] ) {
+				require_lib( 'external-connections' );
+				$connections = \WPCOM_External_Connections::init();
+				return $connections->delete_keyring_connection( $connection['ID'] );
+			}
+			return false;
+		}
+
+		$response = Client::wpcom_json_api_request_as_user(
+			'/me/connections/' . $connection['ID'],
+			'2',
+			array( 'method' => 'DELETE' )
+		);
+		if ( is_wp_error( $response ) ) {
+			return false;
+		}
+		$body = json_decode( wp_remote_retrieve_body( $response ) );
+		return $body->deleted ?? false;
+	}
+
+	/**
+	 * Registers connection settings.
+	 */
+	public static function register_settings() {
+		foreach ( self::$services as $page => $services ) {
```

Comment:
Thinking out loud (nonblocking): I wonder if we could avoid having a foreach and access self::$services[ $your_page_name_key ] directly 🤔.

Reply:
In that case, we'd need to pass the `$page` param to this function, and update the `add_settings_for_service` method to register multiple hooks (one for each page), so we wouldn't change anything essentially.